### PR TITLE
Adds a github action to validate every PR labels and potentially release notes

### PR DESCRIPTION
**Pull request by mohammedDehairy**
Adds a github action to validate every PR labels and potentially release notes.

### CODE REVIEW

File: .github/workflows/cancel-duplicate-work-flows.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Cancel duplicate work flows
```
Comment (descorp) [inline .github/workflows/cancel-duplicate-work-flows.yml:1]: Nice!

Comment (mohammedDehairy) [reply .github/workflows/cancel-duplicate-work-flows.yml:1]: yes, super nice, right?

File: .github/workflows/validate_pr_labels_and_release_notes.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Validate PR labels and release notes
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled, edited]
```
Comment (descorp) [inline .github/workflows/validate_pr_labels_and_release_notes.yml:5]: 👍

File: pull-requests-allowed-labels-list [hunk 3]
```diff
@@ -0,0 +1 @@
+new,changed,fixed, removed
```
Comment (descorp) [inline pull-requests-allowed-labels-list]: Maybe move it to .github folder?
And also filename, maybe `.pull-requests-allowed-labels-list` ?

Comment (descorp) [inline pull-requests-allowed-labels-list]:
```suggestion
new,changed,fixed,removed,deprecated
```

**Comment by descorp** [comment]
What if we use following patterns:

```
## [Label]
* [item]
```
So each release note will be combination of `##` header + at-list one of bullet points `*` ?

**Review by descorp — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Pull request merged**